Sources/MapboxMaps/Foundation/ResourceOptionsManager.swift

```diff
@@ -0,0 +1,151 @@
+    /// for `ResourceOptions` and `MapInitOptions`.
+    ///
+    /// The application's Info.plist will be searched for a valid access token
+    /// under the key `MBXAccessToken`
+    ///
+    /// A valid access token must be provided or found.
+    public static var `default`: ResourceOptionsManager {
+        if defaultInstance == nil {
+            defaultInstance = ResourceOptionsManager(accessToken: nil)
+        }
+        return defaultInstance!
+    }
+
+    /// :nodoc:
+    /// Convenience function to remove the default instance. Calling `default`
+    /// again will re-create the default instance.
+    internal static func destroyDefault() {
+        defaultInstance = nil
+    }
+
+    private static var defaultInstance: ResourceOptionsManager?
+
+    /// Return the current resource options.
+    public var resourceOptions: ResourceOptions {
+        get {
+            return _resourceOptions
+        }
+        set {
+            update(newValue)
+        }
+    }
+
+    private var _resourceOptions: ResourceOptions!
+    private var tileStore: TileStore!
+    private var bundle: Bundle
+
+    /// Initializes a `ResourceOptionsManager` with an optional access token.
+    ///
+    /// If the supplied token is nil (which is the case for the `default`) then
+    /// we will use appropriate defaults for the `ResourceOptions`, including
+    /// searching for an access token in the application's Info.plist.
+    ///
+    /// You can override the shared global access token, using
+    /// `ResourceOptionsManager.default`:
+    ///
+    ///     ```
+    ///     ResourceOptionsManager.default.update { resourceOptions in
+    ///         resourceOptions.accessToken = "overridden-access-token"
+    ///     }
+    ///     ```
+    ///
+    /// - Parameter accessToken: Valid access token or `nil`
+    public convenience init(accessToken: String? = nil) {
+        let resourceOptions = ResourceOptions(accessToken: accessToken ?? "")
+        self.init(resourceOptions: resourceOptions)
+    }
+
+    public convenience init(resourceOptions: ResourceOptions) {
+        self.init(resourceOptions: resourceOptions, for: .main)
+    }
+
+    /// Initializes a `ResourceOptionsManager` with the specified access token
+    /// and `Bundle`.
+    ///
+    /// - Parameters:
+    ///     - accessToken: Valid access token or `nil`
+    ///     - bundle: Bundle to search for an access token (used if resource
+    ///         options is nil).
+    internal init(resourceOptions: ResourceOptions, for bundle: Bundle) {
+        self.bundle = bundle
+        update(resourceOptions)
+    }
+
+    /// Reset the manager to a default `ResourceOptions` using the specified
+    /// access token.
+    ///
+    /// - Parameter accessToken: Valid access token or `nil`
+    internal func reset(resourceOptions: ResourceOptions? = nil) {
+        self.resourceOptions = resourceOptions ?? ResourceOptions(accessToken: defaultAccessToken())
```

Comment:
The default access token logic should probably move to where the core ResourceOptions object is initialized from the struct.